packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -74,29 +79,31 @@ export abstract class TlsValidationTrust {
   public abstract bind(scope: Construct): TlsValidationTrustConfig;
 }
 
+/**
+ * Represents a TLS Validation Context Trust that is supported for mutual TLS authentication.
+ */
+export abstract class MutualTLSAuthEligibleTlsValidationTrust extends TlsValidationTrust {
+  // TypeScript uses structural typing, so we need a property different from TlsValidationTrust
+  protected readonly differentiator = false;
+}
+
 class TlsValidationAcmTrust extends TlsValidationTrust {
```

Comment:
I think it depends on whether there will a possibility of supporting difference kind of certificates other than File, AMC, and SDS.

- If we will only handle above three certificate sources: remove `MutualTLSAuthEligibleTlsValidationTrust` since we no longer need to distinguish
- If we might support other certificates sources: Leave it as it is in case to handle similar scenario